crates/subspace-transaction-pool/src/lib.rs

```diff
@@ -459,15 +548,64 @@ where
     BundleValidator:
         ValidateBundle<Block, domain_runtime_primitives::Hash> + Clone + Send + Sync + 'static,
     Verifier: VerifyFraudProof + Clone + Send + Sync + 'static,
+{
+    let prometheus = config.prometheus_registry();
+    let pool_api = Arc::new(FullChainApiWrapper::new(
+        client.clone(),
+        prometheus,
+        task_manager,
+        FullChainVerifier::new(client.clone(), verifier, bundle_validator),
+    ));
+    let pool = Arc::new(BasicPoolWrapper::with_revalidation_type(
+        config,
+        pool_api,
+        prometheus,
+        task_manager.spawn_essential_handle(),
+        client.clone(),
+    ));
+
+    // make transaction pool available for off-chain runtime calls.
+    client
+        .execution_extensions()
+        .register_transaction_pool(&pool);
+
+    pool
+}
+
+/// Constructs a transaction pool with provided verifier.
+pub fn new_full_with_validator<Block, Client, Verifier>(
+    config: &Configuration,
+    task_manager: &TaskManager,
+    client: Arc<Client>,
+    verifier: Verifier,
```

Comment:
The naming looks inconsistent, `with_validator` but accepts verifier, also in domains/service/src/core_domain.rs L187 `core_domain_xdm_verifier` is constructed from `CoreDomainXDMValidator`.